azkaban-web-server/src/main/java/cloudflow/services/ExecutionServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+package cloudflow.services;
+
+
+import azkaban.executor.ExecutableFlow;
+import azkaban.executor.ExecutableFlowBase;
+import azkaban.executor.ExecutableNode;
+import azkaban.executor.ExecutionAttempt;
+import azkaban.executor.ExecutorManagerAdapter;
+import azkaban.executor.ExecutorManagerException;
+import azkaban.executor.Status;
+import cloudflow.error.CloudFlowError;
+import cloudflow.error.CloudFlowException;
+import cloudflow.error.ErrorCode;
+import cloudflow.models.JobExecution;
+import cloudflow.models.JobExecutionAttempt;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExecutionServiceImpl implements ExecutionService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ExecutionServiceImpl.class);
+    private ExecutorManagerAdapter executorManager;
+
+    @Inject
+    public ExecutionServiceImpl(ExecutorManagerAdapter executorManager) {
+        this.executorManager = executorManager;
+    }
+
+    @Override
+    public Optional<JobExecution> getJobExecution (String executionId, String jobDefinitionId,
```

Comment:
We should look into breaking down this function into smaller functions later on. With all the TODOs it will become quite long.

Reply:
I agree.